src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -410,6 +420,30 @@ type DiagnosticItem =
         with
         | _ -> str // let's fail silently for now
 
+    // TODO: When the names of the runtime capabilities are finalized, they can be included in the result
+    //       comparison error messages.
+    static member private CapabilityErrorFormatString(code) =
+        match code with
+        | ErrorCode.UnsupportedResultComparison -> "The target {0} does not support comparing measurement results."
+        | ErrorCode.ResultComparisonNotInOperationIf ->
+            "Measurement results cannot be compared here. "
+            + "The target {0} only supports comparing measurement results as part of the condition of an if- or elif-statement in an operation."
+        | ErrorCode.ReturnInResultConditionedBlock ->
+            "A return statement cannot be used here. "
+            + "The target {0} does not support return statements in conditional blocks that depend on a measurement result."
+        | ErrorCode.SetInResultConditionedBlock ->
+            "The variable \"{0}\" cannot be reassigned here. "
+            + "In conditional blocks that depend on a measurement result, the target {1} only supports reassigning variables that were declared within the block."
+        | ErrorCode.UnsupportedCallableCapability ->
+            "The callable {0} requires runtime capabilities which are not supported by the target {1}."
+            + Environment.NewLine
+            + "Result Opacity: {2}"
+            + Environment.NewLine
+            + "Classical Capability: {3}"
+        | ErrorCode.UnsupportedClassicalCapability ->
+            "This construct requires a classical runtime capability that is not supported by the target {0}: {1}."
```

Comment:
(nit) Consider
```suggestion
        | ErrorCode.UnsupportedClassicalCapability ->
            "This construct requires a classical runtime capability {1} that is not supported by the target {0}."
```

Reply:
It would be interesting to see @samarsha opinion on this. These messages were already present in the code, they were already approved by other reviewers, I just moved them to a different place in the file. The proposed change makes it sound better, but there could be other reasons to phrase the way it is.